File: backend/src/main/java/com/stijaktech/devnews/models/Post.java
package com.stijaktech.devnews.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;

import java.time.Instant;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {

    @Id
    private String id;

    private String title;

    private String content;

    @DBRef
    @CreatedBy
    // @JsonIgnore
    private User createdBy;

    @DBRef
    // @JsonIgnore
    @LastModifiedBy
    private User updatedBy;

    @CreatedDate
    private Instant createdAt;

    // @JsonIgnore
    @LastModifiedDate
    private Instant updatedAt;

    private List<Comment> comments;

}

File: frontend/src/app/models/post.ts
import {User} from './user';
import {Community} from './community';
import {Links} from './hal';
import {Comment} from './comment';

export class Post {

  public id: string;
  public alias: string;
  public title: string;
  public content: string;
  public createdBy: User;
  public createdAt: string;
  public updatedAt: string;
  public comments: Comment[];
  public commentsCount: number = 0;
  public community: Community;
  public _links: Links;

}
